core/src/main/java/org/testcontainers/containers/ContainerDef.java

```diff
@@ -0,0 +1,253 @@
+package org.testcontainers.containers;
+
+import com.github.dockerjava.api.command.CreateContainerCmd;
+import com.github.dockerjava.api.model.Bind;
+import com.github.dockerjava.api.model.ExposedPort;
+import com.github.dockerjava.api.model.HostConfig;
+import com.github.dockerjava.api.model.InternetProtocol;
+import com.github.dockerjava.api.model.PortBinding;
+import com.github.dockerjava.api.model.Ports;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.testcontainers.DockerClientFactory;
+import org.testcontainers.UnstableAPI;
+import org.testcontainers.containers.wait.strategy.WaitStrategy;
+import org.testcontainers.images.RemoteDockerImage;
+import org.testcontainers.utility.Base58;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.TestcontainersConfiguration;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+@Getter
+@UnstableAPI
+@Slf4j
+class ContainerDef {
+
+    private RemoteDockerImage image;
+
+    private Set<ExposedPort> exposedPorts = new HashSet<>();
+
+    private Set<PortBinding> portBindings = new HashSet<>();
+
+    private Map<String, String> labels = new HashMap<>();
+
+    private Map<String, String> environmentVariables = new HashMap<>();
+
+    private String[] command = new String[0];
+
+    private Network network;
+
+    private Set<String> networkAliases = new HashSet<>(Collections.singletonList("tc-" + Base58.randomString(8)));
+
+    private String networkMode;
+
+    private List<Bind> binds = new ArrayList<>();
+
+    private boolean privilegedMode;
+
+    private WaitStrategy waitStrategy;
+
+    private boolean frozen = false;
+
+    public ContainerDef() {}
+
+    private ContainerDef(ContainerDef other) {
```

Comment:
Can we use Lombok to generate it? It looks like `waitStrategy` is missing, and I am worried about us forgetting to include more fields in the future